File: pow9/pow9.py
board = []
for row in xrange(0, 1000):
    board.append([])
    for column in xrange(0, 100):
        board[row].append(0)

def check_left(row,column,board):
    numbers_left = []
    for iter_column in xrange(0, column):
        numbers_left.append(board[row][iter_column])

    if board[row][column] in numbers_left:
        return False
    else:
        return True

def check_up(row,column,board):
    numbers_up = []
    for iter_row in xrange(0, row):
        numbers_up.append(board[iter_row][column])

    if board[row][column] in numbers_up:
        return False
    else:
        return True

for row in xrange(0,1000):
    for column in xrange(0, 100):
        while(not check_left(row, column, board) or not check_up(row, column, board)):
            board[row][column] += 1

print board

File: pow9/pow9.rb
board = []
(0..50).each do |row|
	board[row] = []
	(0..50).each do |column|
		board[row][column] = 0
	end
end

def check_left(row, column, board)
	numbers_left = (0..column-1).collect { |iter_column| board[row][iter_column]}

	if numbers_left.include?(board[row][column])
		return false
	else
		return true
	end
end

def check_up(row, column, board)
	numbers_up = (0..row-1).collect { |iter_row| board[iter_row][column]}

	if numbers_up.include?(board[row][column])
		return false
	else
		return true
	end
end

(0..50).each do |row|
	(0..50).each do |column|
		while ( !check_left(row, column, board) or !check_up(row, column, board))
			board[row][column] += 1
		end
	end
end

puts board.inspect